prow/config/secrets_agent.go

```diff
@@ -31,9 +31,20 @@ type SecretAgent struct {
 	secretsMap map[string][]byte
 }
 
-// Start will begin polling the secret file at the path. If the first load
-// fails, Start with return the error and abort. Future load failures will log
-// the failure message but continue attempting to load.
+// NewSecretAgent creates a SecretAgent. This can be used if there is no need to monitor
+// files to update the secret value. Useful for testing.
+func NewSecretAgent(tokenKeys []string) *SecretAgent {
```

Comment:
It is strange to accept tokenKeys that are set to dummy values that are invalid. The SetSecret function will always need to be called afterwards to provide a valid secret value.

I think the API would be better if NewSecretAgent just initialized an empty secretMap and didnt take any args. This makes it clear that SetSecret needs to be called afterwards and avoids storing invalid secret values.